sdk/monitor/monitor-query/review/monitor-query.api.md

```diff
@@ -9,9 +9,26 @@ import { OperationOptions } from '@azure/core-client';
 import { PagedAsyncIterableIterator } from '@azure/core-paging';
 import { TokenCredential } from '@azure/core-auth';
 
+// @public
+export class AggregateBatchError extends Error {
+    constructor(errors: ErrorInfo[]);
+    // (undocumented)
+    errors: BatchError[];
```

Comment:
do we need `BatchError` in the public surface? `ErrorInfo` already extends `Error` and contains everything manually specified in `BatchError`. I think we could say that this is just an array of `ErrorInfo` (even if they really are BatchError underneath)

Reply:
though for the single query operation in case of error we are returning `BatchError` and for batch queries we are returning errors of type `AggregateBatchError`.

Reply:
Right but `BatchError` has a public shape that looks identical to `ErrorInfo`? So why have a class and an interface that are the same? We don't expect users to call `new` on BatchError, do we?

Reply:
so `BatchError` extends `Error` - `ErrorInfo` doesn't - with that we get the proper Error type that we can use to throw to users, with the `name` property - `ErrorInfo` doesn't have `name`

Reply:
Users won't call `new` on this. But they can use that type to understand the return type

Reply:
we couldn't use `ErrorInfo` directly to throw the error object - had to convert it to extend from `Error` class so the shape of the error thrown is according to our guidelines

Reply:
It does look like ErrorInfo extends Error as well https://github.com/Azure/azure-sdk-for-js/pull/17562/files#diff-a9aa04e6cc40074ee7920974c697661890fa83c351bb892ce7c0108927a1b800R57 

Maybe that's changed recently? 

The two look the same besides the constructor - so we should revisit whether BatchError is even needed. Is it because ErrorInfo is an interface and BatchError is a class and you want to `new` one? 

I do agree with @xirzec that one is superfluous though... so we should be able to consolidate here